packages/grid/src/vaadin-grid.js

```diff
@@ -509,13 +509,36 @@ class Grid extends ElementMixin(
       reorderElements: true,
     });
 
-    new ResizeObserver(() =>
+    // Memorize last scroll position in Firefox
+    if (this._firefox) {
+      this.$.table.addEventListener('scroll', () => {
+        const isVisible = !isElementHidden(this);
+        if (isVisible && this.__previousVisible !== false) {
+          this.__memorizedScrollTop = this.$.table.scrollTop;
+        }
+      });
+    }
+
+    new ResizeObserver(() => {
+      // For Firefox, manually restore last scroll position when grid becomes
+      // visible again. This solves an issue where switching visibility of two
+      // grids causes Firefox trying to synchronize the scroll positions between
+      // the two grid's table elements.
+      // See https://github.com/vaadin/web-components/issues/5796
+      if (this._firefox) {
+        const isVisible = !isElementHidden(this);
+        if (isVisible && this.__previousVisible === false) {
+          this.$.table.scrollTop = this.__memorizedScrollTop || 0;
+        }
+        this.__previousVisible = isVisible;
+      }
```

Comment:
Since this logic is scroll-related, `vaadin-grid-scroll-mixin` could be a nicer place for it. There's also an existing `_afterScroll` method in the mixin that could be used.

Reply:
Thanks, was able to move this into the existing `_afterScroll` and `_onResize` methods.